spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java

```diff
@@ -113,17 +119,12 @@ public RootController rootController(EntityLinks entityLinks) {
 	}
 
 	@Bean
-	public RuntimeAppsController runtimeAppsController(StreamDefinitionRepository repository,
-			DeploymentIdRepository deploymentIdRepository, AppDeployer appDeployer) {
-		return new RuntimeAppsController(repository, deploymentIdRepository, appDeployer);
-	}
-
-	@Bean
 	public AppInstanceController appInstanceController(AppDeployer appDeployer) {
 		return new AppInstanceController(appDeployer);
 	}
 
 	@Bean
+	@ConditionalOnBean(StreamDefinitionRepository.class)
```

Comment:
If those can only work with some collaborators (which are driven by feature flags) then, in some cases at least, doesn't it make sense to declare in that same configuration class (that is driven by the flag)?